isis/src/socet/apps/socetframesettings/socetframesettings.cpp

```diff
@@ -471,110 +683,284 @@ void getCamPosOPK(Spice &spice, QString spacecraftName, double et, Camera *cam,
+                    |      | \                                         |
+                    |      |    *                                      |
+                    |      |           |     ___                       |
+                    |      |    |    --+--  /   \                      |
+                    |    --+--  |      |    \___                       |
+                    |      |    |      |        \                      |
+                    |      |    |_/    |_/  \___/                      |
+                    |                                                  |
+                    |                                                  |
+                    |       _    ___    _    ___                       |
+                    |      | \  /   \  | \  /   \                      |
+                    |      |     ___/  |     ___/                      |
+                    |      |    /      |    /                          |
+                    |      |    \___/  |    \___/                      |
+                    |                                                  |
+ +NAXIS2fits ^      |      *           *                               |
+ +LINEisis   |      |                                                  |
+ +Xfits      |      |                                                  |
+ +Xisis      |      |(1,1)isis == (0,0)fits                            |
+             |      X--------------------------------------------------+
+             |
+             |
+             |+Zfits = +Zisis (into screen)
+          ---X---------------------------------> +NAXIS1fits
+             |                                   +SAMPLEisis
+             |                                   +Yfits      
+             |                                   +Yisis      
+
+ +Xss == +Yisis == +Yfits
+ +Yss == +Xisis == +Xfits
+ +Zss == -Zisis == -Zfits
+
+   ********************************************************************/
+
+  if (spacecraftName == "OSIRIS-REX") {
+    /* MapCam and PolyCam ISIS-to-SS Matrix swaps X and Y, inverts Z */
+    isisFocalPlane2SocetPlate[1][0] =  1.0;  // +Xisis => +Yss
+    isisFocalPlane2SocetPlate[0][1] =  1.0;  // +Yisis => +Xss
+    isisFocalPlane2SocetPlate[2][2] = -1.0;  // +Zisis => -Zss
+  }
+
+  // Confirm that matrix is now a rotation matrix
+  if (!isrot_c(isisFocalPlane2SocetPlate, 1e-10, 1e-10)) {
+    QString msg = QString("The ISIS to SOCET Set translation of input image is currently "
+                          "not supported for instrument [%1].").arg(spacecraftName);
+    throw IException(IException::User, msg, _FILEINFO_);
+  }
+
+  //////////////////////////////////////////////////////////////////////
+  //////////////////////////////////////////////////////////////////////
+  // End of section setting isisFocalPlane2SocetPlate
+  //////////////////////////////////////////////////////////////////////
+  //////////////////////////////////////////////////////////////////////
```

Comment:
If this section is so large that the end of it needs to be indicated, it should be moved to its own method.

Reply:
Or just moved to the end of the file with a note referencing it here?